aten/src/ATen/native/native_functions.yaml

```diff
@@ -10927,7 +10927,11 @@
   dispatch:
     CPU: searchsorted_cpu
     CUDA: searchsorted_cuda
-  autogen: searchsorted.Scalar_out
+
+- func: searchsorted.Scalar_out(Tensor sorted_sequence, Scalar self, *, bool out_int32=False, bool right=False, str? side=None, Tensor? sorter=None, Tensor(a!) out) -> Tensor(a!)
+  dispatch:
+    CPU: searchsorted_out_cpu
+    CUDA: searchsorted_out_cuda
```

Comment:
I'll take a closer look at this. Looks like autogen doesn't generate the out variant. Maybe I'll fix the codegen instead.

```py
In [42]: torch.searchsorted(xs, torch.tensor(2), out=torch.tensor((), dtype=torch.int64))
Out[42]: tensor(6)

In [43]: torch.searchsorted(xs, 2, out=torch.tensor((), dtype=torch.int64))
TypeError: searchsorted() received an invalid combination of arguments - got (Tensor, int, out=Tensor), but expected one of:
 * (Tensor sorted_sequence, Tensor input, *, bool out_int32, bool right, str side, Tensor sorter, Tensor out)
 * (Tensor sorted_sequence, Number self, *, bool out_int32, bool right, str side, Tensor sorter)
```

Reply:
I've looked at this a bit. It seems it's a known issue with autogen: it does generate things on the aten side, but not on the Python side.

```py
    # A list of operators that are expected to be auto-generated for this NativeFunction.
    # Note: This list isn't actually directly used by the codegen to generate anything.
    # Instead, the codegen figures out what operators to generate purely based off of
    # function schema, and uses the autogen declarations to error check.
    # We expect every NativeFunction that gets auto-generated be explicitly called out
    # in native_functions.yaml
    autogen: List["OperatorName"]
```

So it seems like a serious undertaking to fix this on the codegen side. I propose we just add out variants to unblock this.